lib/mongo/cluster.rb

```diff
@@ -497,13 +497,14 @@ def summary
     # events in the process. Stops SRV monitoring if it is active.
     # Marks the cluster disconnected.
     #
-    # @return [ true ] Always true.
+    # A closed cluster is no longer usable. If the client is reconnected,
+    # it will create a new cluster instance.
     #
-    # @since 2.1.0
-    def disconnect!
```

Comment:
Leave the existing `disconnect!` method, mark as deprecated
Add the new method `close` method, and use it when we should.

Reply:
I think we can actually leave this and mark the new method @api private.... See the note on this method:

```
    # @note Applications should call Client#close to disconnect from
    # the cluster rather than calling this method. This method is for
    # internal driver use only.
```

Reply:
Let me know if you still want me to add it back.